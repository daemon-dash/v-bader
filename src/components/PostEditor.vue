<template>
  <div class="container">
    <div class="centered">
      <input
        v-model="title"
        type="text"
        class="title"
        placeholder="title"
      /><br />
      <textarea v-model="body" class="content" rows="3" placeholder="content">
      </textarea>
      <br />
      <div class="buttons">
        <!--
          trigered on click
          @event click
        -->
        <button @click="onEditDone" class="done">Done</button>
      </div>
    </div>
  </div>
</template>
<script>
export default {
  /**
   * the values of title and body will be passed down to component
   * @binding {title} title is currentTitle in state
   * @binding {body} content text of current Post
   */
  props: ["post"],
  data() {
    return {
      title: this.post.title,
      body: this.post.body
    };
  },
  methods: {
    onEditDone() {
      if (this.title !== "" || this.body !== "") {
        this.$emit("editDone", {
          title: this.title,
          body: this.body
        });
      }

      this.title = "";
      this.body = "";
    }
  }
};
</script>
<style scoped>
.title,
.content {
  border: none;
  width: calc(100% - 20px);
  padding: 10px;
  display: inline-block;
}
.title {
  font-weight: bold;
}
.container {
  text-align: center;
  margin-top: 0;
}
.container .centered {
  display: inline-block;
  width: 80%;
  text-align: left;
  background-color: white;
  box-shadow: 3px 3px 5px grey;
}
.buttons {
  padding: 10px;
}
</style>
