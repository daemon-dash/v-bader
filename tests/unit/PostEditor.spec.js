import Vuex from "vuex";
import { shallowMount, createLocalVue } from "@vue/test-utils";
import PostEditor from "../../src/components/PostEditor.vue";

const localVue = createLocalVue();
localVue.use(Vuex);

describe("PostEditor.vue", () => {
  it("Shallow renders Post Editor", () => {
    const wrapper = shallowMount(PostEditor, {
      propsData: {
        post: {
          title: "Test Data",
          body: "Test body",
        },
      },
    });

    expect(wrapper.html()).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
