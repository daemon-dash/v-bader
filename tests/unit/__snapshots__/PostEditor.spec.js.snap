// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostEditor.vue Shallow renders Post Editor 1`] = `
<div class="container">
  <div class="centered"><input type="text" placeholder="title" class="title"><br> <textarea rows="3" placeholder="content" class="content"></textarea> <br>
    <div class="buttons"><button class="done">Done</button></div>
  </div>
</div>
`;
